javalib/src/main/scala/java/net/Net.scala

```diff
@@ -0,0 +1,286 @@
+package java.net
+
+import java.io.FileDescriptor
+import java.nio.channels.UnsupportedAddressTypeException
+import scala.scalanative.meta.LinktimeInfo.isWindows
+import scala.scalanative.posix
+import scala.scalanative.posix.netinet.{in, tcp}
+import scala.scalanative.posix.sys.socketOps._
+import scala.scalanative.unsafe._
+import scala.scalanative.unsigned._
+
+trait Net {
+  import Net._
+
+  def POLLIN: Int
+  def POLLOUT: Int
+
+  def prepareSockaddrIn(
+      isa: InetSocketAddress
+  )(implicit
+      zone: Zone
+  ): (Ptr[posix.sys.socket.sockaddr], posix.sys.socket.socklen_t) = {
```

Comment:
Allocating the `sockaddr` on heap can allow to factorize a lot of code (`bind4/bind6`, `connect4/connect6`, ...). Is this something allowed? Or should we stick with stack allocation only ?

Reply:
It's completely fine to use heap allocations. No need to limit ourself, especially when stuff is getting more complex.

Reply:
I  concur. I think the `stackalloc` was done at the beginning
to reduce the number of allocations. It then propagated to 
IPv6 and datagrams.  

Given that 'over-the-wire' time is likely to vastly dominate network code, a few judicious extra allocations are probably not going to move the meter, as much as I like short code paths.